### PR TITLE
Python: fix "no such file or directory" job exception resulting from bad jobspec

### PR DESCRIPTION
**Pull request by jameshcorbett**
Two pretty trivial changes. The first commit fixes a problem I introduced in #3019. When you set a jobspec's stdout/err property to ``None``, the job will hit an exception:

```
JobWaitResult(jobid=22162702336, success=False, errstr=b'Fatal exception type=exec shell_init: No such file or directory')
```

The error message actually confused me for a little while, and it's such an easy fix, that I think it's worth it.

(Edit: it's especially confusing because the properties default to ``None``, which works fine, but if you explicitly set them to ``None`` the job will fail.)

The second commit removes references to the ``six`` module. I noticed them while I was working on the first commit and so I just went ahead and replaced the ``six`` aliases with their Python 3 names.

I would be happy to squash these two commits down to one if you want, just let me know.

### CODE REVIEW

File: src/bindings/python/flux/core/handle.py [hunk 1]
```diff
@@ -11,7 +11,6 @@
 import signal
 import threading
 from contextlib import contextmanager
-import six
```
Comment (SteVwonder) [inline src/bindings/python/flux/core/handle.py:14]: Does this commit remove every reference to `six` in the codebase?  If so, can you remove the check for `six` in the configure.ac: https://github.com/flux-framework/flux-core/blob/master/configure.ac#L253

And also remove `six` from the dependency table in the README.md: https://github.com/flux-framework/flux-core/blame/master/README.md#L59

File: src/bindings/python/flux/job/Jobspec.py [hunk 2]
```diff
@@ -389,6 +388,10 @@ def _set_io_path(self, iotype, stream_name, path):
         :param stream_name: the name of the io stream
         :param path: the path to redirect the stream
         """
+        if isinstance(path, os.PathLike):
+            path = os.fspath(path)
+        if not isinstance(path, str):
+            raise TypeError("The path must be a string")
```
Comment (SteVwonder) [inline src/bindings/python/flux/job/Jobspec.py:394]:
```suggestion
            raise TypeError("The path must be a string or pathlib object")
```

**Review by SteVwonder — COMMENTED** [review]
Thanks @jameshcorbett!

**Comment by SteVwonder** [comment]
> I would be happy to squash these two commits down to one if you want, just let me know.

I think it makes sense to leave them as two commits.  I'll defer to others on that decision though.

File: src/bindings/python/flux/core/handle.py (hunk 1 above)
Comment (jameshcorbett) [reply src/bindings/python/flux/core/handle.py:14]: Thanks! I would never have thought of that. I got rid of all the other references to six and then fixed up the README.md and configure.ac.

**Review by SteVwonder — APPROVED** [review]
Thanks @jameshcorbett! LGTM!

**Comment by grondo** [comment]
> I think it makes sense to leave them as two commits. I'll defer to others on that decision though.

Separate commits is perfect, thanks!

**Comment by grondo** [comment]
Thank you *so much* for cleaning up that `six` stuff! I kept wanting to do it, but wasn't sure exactly how.

**Comment by grondo** [comment]
This looks good to me. I'm going to set MWP unless you have more you want to do here @jameshcorbett

**Comment by jameshcorbett** [comment]
> This looks good to me. I'm going to set MWP unless you have more you want to do here @jameshcorbett

Nope, thanks!

**Pull request merged**